### PR TITLE
(Chore) Refactor CheckboxList component

### PR DESCRIPTION
**Pull request by janjaap**
This PR simplifies the use of the `CheckboxList` component. To clarify its purpose, separate documentation has been added. See [README.md](https://github.com/Amsterdam/signals-frontend/blob/417604205ac05af944d2f95a500d9fb44f24d229/src/signals/incident-management/components/CheckboxList/doc/README.md) for reference.

### CODE REVIEW

File: src/signals/incident-management/components/CheckboxList/doc/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    { key: 'A', value: 'Centrum' },
+    { key: 'B', value: 'Westpoort' },
+  ]}
+  title="Stadsdeel"
+/>
+```
+
+will render:
+
+<img src="./toggle.png" alt="" width="220" />
```
Comment (spaces-ga) [inline src/signals/incident-management/components/CheckboxList/doc/README.md]: Looks the same...

File: src/signals/incident-management/components/FilterForm/index.js [hunk 2]
```diff
@@ -82,19 +82,22 @@ const FilterForm = ({
...
-    const btnHasSaveLabel = submitBtnLabel === saveSubmitBtnLabel;
+      const formData = parseOutputFormData(event.form);
+      const valuesHaveChanged = !isEqual(formData, filterData);
+      const btnHasSaveLabel = submitBtnLabel === saveSubmitBtnLabel;
 
-    if (valuesHaveChanged && !btnHasSaveLabel) {
-      setSubmitBtnLabel(saveSubmitBtnLabel);
-    }
-  }, [filterData, isNewFilter, submitBtnLabel]);
+      if (valuesHaveChanged && !btnHasSaveLabel) {
```
Comment (spaces-ga) [inline src/signals/incident-management/components/FilterForm/index.js]: Is this necessary in the callback? Can't this be checked in the render step? Looks like changing the button label will cause an extra rerender now.

File: src/signals/incident-management/components/CheckboxList/doc/README.md (hunk 1 above)
Comment (janjaap) [reply src/signals/incident-management/components/CheckboxList/doc/README.md]: It does, yes. The HTML, though, is different.

File: src/signals/incident-management/components/FilterForm/index.js (hunk 2 above)
Comment (janjaap) [reply src/signals/incident-management/components/FilterForm/index.js]: No, has to be in the `onChangeForm` callback; the value of the form fields has to be compared to those that are in the state after something has changed in the form. That't not possible by moving it to the render step.

**Pull request merged**